src/GrasshopperTeklaDrawingLink/Components/Text/TextAttributesComponent.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Grasshopper.Kernel;
+
+using GTDrawingLink.Properties;
+using GTDrawingLink.Tools;
+using GTDrawingLink.Types;
+
+using System.Collections.Generic;
+using System.Drawing;
+
+using Tekla.Structures.Drawing;
+
+using TSD = Tekla.Structures.Drawing;
+
+namespace GTDrawingLink.Components.Text {
+    public class TextAttributesComponent : TeklaComponentBase {
+        public override GH_Exposure Exposure => GH_Exposure.primary;
+        protected override Bitmap Icon => Resources.LineTypeAttributes;
+
+        public TextAttributesComponent()
+            : base(ComponentInfos.TextAttributesComponent) {
+        }
+
+        protected override void RegisterInputParams(GH_InputParamManager pManager) {
+            List<int> indeces = new List<int> {
+                pManager.AddParameter(new FontAttributesParam(ParamInfos.FontAttributes, GH_ParamAccess.item)),
+                pManager.AddParameter(new EnumParam<FrameTypes>(ParamInfos.FrameType, GH_ParamAccess.item)),
+                pManager.AddParameter(new EnumParam<DrawingColors>(ParamInfos.DrawingColor, GH_ParamAccess.item)),
+                 pManager.AddParameter(new ArrowAttributesParam(ParamInfos.ArrowAttribute, GH_ParamAccess.item))
+            };
+            SetParametersAsOptional(pManager, indeces);
```

Comment:
Different style of using _SetParametersAsOptional_ method. Previous attempts for injecting parameter declarations as an input for _SetParametersAsOptional_ was more clear in reading.